### PR TITLE
ANW-2617 Add an AppConfig option to set the maximum allowable rows th…

### PR DESCRIPTION
**Pull request by lorawoodford**
…at will be accepted by the Bulk Archival Object Updater job

## Description

Adds an new AppConfig - `AppConfig[:bulk_archival_object_updater_max_rows]` - that allows systems administrators to set a maximum number of rows that will be accepted in a Bulk Archival Object Updater job spreadsheet.  

I have made several judgement calls here that should get careful review from the ASpace team:
1. I've set the default max_rows to 1,000, but this is based on our own institutional testing.  It is around that number of records that we start to see declined performance and timeouts appearing in our logs.  But that's just us -- greater thought should probably go into this.  I will say, I discovered while testing this against latest master, that the rack bump that was brought in via https://github.com/archivesspace/archivesspace/pull/3850 adds some urgency to this change (or a larger refactor to the bulk updater).  [The new 4096 param limit in rack](https://redirect.github.com/rack/rack/commit/3f5a4249118d09d199fe480466c8c6717e43b6e3) will lead to spreadsheets above 4000ish failing to be generated altogether.  This is what I get when I attempt generate a sheet with around 7K rows (which is not all that unusual for our holdings here):
Screenshot 2026-02-11 at 12 06 32 PM

2. The language - both in the background job error and the warning - should of course be reviewed.
3. In adding the e2e tests I had to get creative with how to adjust an AppConfig setting on the fly.  It's possible as the e2e tests mature you may want to come up with a better way to test these configuration states.

## Related JIRA Ticket or GitHub Issue

https://archivesspace.atlassian.net/browse/ANW-2617

## How Has This Been Tested?

Tests added and passing.

## Screenshots (if appropriate):
When generating a spreadsheet that exceeds the setting, you get a warning:
Screenshot 2026-02-10 at 3 56 38 PM

When attempting to upload a sheet that is too large:
Screenshot 2026-02-10 at 4 33 55 PM

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have authority to submit this code.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

File: backend/spec/lib_bulk_archival_object_updater_spec.rb [hunk 1]
```diff
@@ -193,8 +193,11 @@
       it 'ensures that the app config has default values and can be successfully read from bulk updater' do
         bulk_archival_object_updater_apply_deletes = AppConfig[:bulk_archival_object_updater_apply_deletes]
         bulk_archival_object_updater_create_missing_top_containers = AppConfig[:bulk_archival_object_updater_create_missing_top_containers]
+        bulk_archival_object_updater_max_rows = AppConfig[:bulk_archival_object_updater_max_rows] = 1000
```
Comment (thimios) [inline backend/spec/lib_bulk_archival_object_updater_spec.rb]: I guess you want to remove the last part of this line ` = 1000` in order to really test whether the config option is set

File: backend/spec/lib_bulk_archival_object_updater_spec.rb [hunk 2]
```diff
@@ -563,6 +566,54 @@
         expect(total_records_count_before).to eq total_records_count_after
       end
     end
+
+    context 'because the number of ao rows in the sheet exceeds that set in AppConfig[:bulk_archival_object_updater_max_rows]', :disable_database_transaction do
```
Comment (thimios) [inline backend/spec/lib_bulk_archival_object_updater_spec.rb]: just to keep the wording uniform, I would replace 'because' with 'when'

File: e2e-tests/staff_features/resources/step_definitions/resource_generate_bulk_spreadsheet.rb [hunk 3]
```diff
@@ -10,11 +10,29 @@
...
-Then 'the user selects the Archival Object on the Generate Bulk Archival Object Spreadsheet page' do
+Then 'the user selects all Archival Objects on the Generate Bulk Archival Object Spreadsheet page' do
   input_fields = all('#bulk_archival_object_updater_table input')
 
   expect(input_fields.length).to eq 2
-  input_fields[1].click
+  input_fields[0].click
+end
+
+When 'AppConfig[:bulk_archival_object_updater_max_rows] is overridden by localStorage' do
```
Comment (thimios) [inline e2e-tests/staff_features/resources/step_definitions/resource_generate_bulk_spreadsheet.rb]: We are trying to keep the e2e test suite step descriptions non-technical so that non-programmers will eventually feel comfortable working with them.

I thought about ways to make this step sound less technical and removing the 'APPCONFIG_MAX_ROWS is removed from localStorage' step by putting the code that removes the APPCONFIG from local storage in the before block of the suite: https://github.com/archivesspace/archivesspace/blob/master/e2e-tests/staff_features/shared/step_definitions.rb#L4 or in an ensure_.... method within this block: https://github.com/archivesspace/archivesspace/blob/master/e2e-tests/staff_features/shared/step_definitions.rb#L12

But I think the most appropriate solution is to avoid adding this kind of edge case in the e2e test suite and implement this as a feature spec instead. What do you think @lorawoodford ?

File: frontend/config/locales/en.yml [hunk 4]
```diff
@@ -1958,6 +1958,12 @@ en:
     minimum_notes_description: The minimum number of already extant note subrecords of each type to be included.
     download_spreadsheet: Download Spreadsheet
     download_spreadsheet_modal_message: Generating this spreadsheet may take a few moments.
+    too_many_rows_warning:
```
Comment (thimios) [inline frontend/config/locales/en.yml:1962]: We try to add translations for all languages when we add new keys (google translate is fine for that).

**Review by thimios — CHANGES_REQUESTED** [review]
Great work! Some changes suggested regarding spec implementation.

File: backend/spec/lib_bulk_archival_object_updater_spec.rb (hunk 1 above)
Comment (lorawoodford) [reply backend/spec/lib_bulk_archival_object_updater_spec.rb]: thanks for catching - should be resolved.

File: backend/spec/lib_bulk_archival_object_updater_spec.rb (hunk 2 above)
Comment (lorawoodford) [reply backend/spec/lib_bulk_archival_object_updater_spec.rb]: Fixed

File: e2e-tests/staff_features/resources/step_definitions/resource_generate_bulk_spreadsheet.rb (hunk 3 above)
Comment (lorawoodford) [reply e2e-tests/staff_features/resources/step_definitions/resource_generate_bulk_spreadsheet.rb]: Thanks!  I've backed out these changes and added to feature specs instead.  I made the call to break out bulk updater related specs (there were already quite a few) into a new `frontend/spec/features/bulk_archival_object_updater_spec.rb` and remove existing specs from `frontend/spec/features/resources_spec.rb`).

File: frontend/config/locales/en.yml (hunk 4 above)
Comment (lorawoodford) [reply frontend/config/locales/en.yml:1962]: Translations added for all languages that already had a `bulk_archival_object_update_spreadsheet` section.

**Review by thimios — APPROVED** [review]
